common4j/src/main/com/microsoft/identity/common/java/interfaces/IKeyPairStorage.java

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.java.interfaces;
+
+import lombok.NonNull;
+
+/**
+ * An interface for a KeyPair storage.
+ * */
+public interface IKeyPairStorage {
+    /**
+     * Gets a value from the storage.
+     *
+     * @param key A key associated to the value.
+     */
+    String get(@NonNull String key);
+
+    /**
+     * Puts a value into the storage.
+     *
+     * @param key A key associated to the value.
+     * @param value value to be persisted.
+     */
+    void put(@NonNull String key, String value);
+}
```

Comment:
Do we need other methods such as `remove(key)` and `contains(key)` - we can probably add them in later I guess when they are needed

Reply:
Yes. We might, but it's not needed now.